From 973507c6ba84008b80423636989320f198ceeaf5 Mon Sep 17 00:00:00 2001
From: Sunil Goutham <sgoutham@marvell.com>
Date: Wed, 30 Oct 2019 11:45:16 +0530
Subject: [PATCH 397/767] octeontx2-pf: Advertise lockless transmit capability

commit 39c012db6c348c4f0d02d4a2724607a4e3fe25b4 from
git@git.assembla.com:cavium/WindRiver.linux.git

Usage of NETIF_F_LLTX is deprecated upstream hence this
patch cannot be upstreamed. But still enabling this locally
as it improves IP forwarding performance especially when all
24 cores are transmitting pkts.

Change-Id: I33a58c0e71d63ee0937ba50c47c5476ba62be094
Signed-off-by: Sunil Goutham <sgoutham@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/18065
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index e87a5a2cab68..cbb4b4ec7059 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -2286,7 +2286,7 @@ static int otx2_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 				NETIF_F_HW_VLAN_CTAG_RX |
 				NETIF_F_HW_VLAN_STAG_RX;
 
-	netdev->features |= netdev->hw_features;
+	netdev->features |= netdev->hw_features | NETIF_F_LLTX;
 
 	netdev->hw_features |= NETIF_F_LOOPBACK | NETIF_F_NTUPLE |
 			       NETIF_F_RXALL;
-- 
2.31.1

